plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/impl/ProxyStreamProducer.java

```
@@ -0,0 +1,122 @@
+ * This is useful when stream is not present locally and needs to be fetched from a node
+ * retrieved using {@link StreamTicket#getNodeId()} where it is present.
+ */
+public class ProxyStreamProducer implements StreamProducer<VectorSchemaRoot, BufferAllocator> {
+
+    private final StreamReader<VectorSchemaRoot> remoteStream;
+
+    /**
+     * Constructs a new ProxyStreamProducer instance.
+     *
+     * @param remoteStream The remote FlightStream to be proxied.
+     */
+    public ProxyStreamProducer(StreamReader<VectorSchemaRoot> remoteStream) {
+        this.remoteStream = remoteStream;
+    }
+
+    /**
+     * Creates a VectorSchemaRoot for the remote FlightStream.
+     * @param allocator The allocator to use for creating vectors
+     * @return A VectorSchemaRoot representing the schema of the remote FlightStream
+     */
+    @Override
+    public VectorSchemaRoot createRoot(BufferAllocator allocator) {
+        return remoteStream.getRoot();
+    }
+
+    /**
+     * Creates a BatchedJob
+     * @param allocator The allocator to use for any additional memory allocations
+     */
+    @Override
+    public BatchedJob<VectorSchemaRoot> createJob(BufferAllocator allocator) {
+        return new ProxyBatchedJob(remoteStream);
+    }
+
+    /**
+     * Returns the deadline for the remote FlightStream.
+     * Since the stream is not present locally, the deadline is set to -1. It piggybacks on remote stream expiration
+     * @return The deadline for the remote FlightStream
+     */
+    @Override
+    public TimeValue getJobDeadline() {
+        return TimeValue.MINUS_ONE;
+    }
+
+    /**
+     * Provides an estimate of the total number of rows that will be produced.
+     */
+    @Override
+    public int estimatedRowCount() {
+        return remoteStream.getRoot().getRowCount();
+    }
+
+    /**
+     * Task action name
+     */
+    @Override
+    public String getAction() {
+        // TODO get it from remote flight stream
```

Do you want to throw `UnsupportedOperationException("Not implemented yet")` here instead of returning empty string?